src/wasm/kotlin/src/arcs/Particle.kt

```diff
@@ -39,16 +39,17 @@ abstract class Particle : WasmObject() {
         onHandleSync(handle, toSync.isEmpty())
     }
 
-    open fun onHandleUpdate(handle: Handle) {}
-    open fun onHandleSync(handle: Handle, allSynced: Boolean) {}
+    open fun init() {}
+    open fun onHandleUpdate(handle: Handle) = renderOutput()
+    open fun onHandleSync(handle: Handle, allSynced: Boolean) = renderOutput()
 
     fun renderOutput() {
         log("renderOutput")
         val slotName = ""
         val template = getTemplate(slotName)
         val dict = populateModel(slotName)
-        val model = StringEncoder.encodeDictionary(dict)
-        onRenderOutput(toWasmAddress(), template.toWasmString(), model.toWasmString())
+        val model = dict?.let { StringEncoder.encodeDictionary(dict) }
```

Comment:
Could drop the `dict` variable and inline `populateModel(slotName)` here.  (and use `it` within the `let` block)